src/OpenTelemetry.AutoInstrumentation/Util/ActivityHelper.cs

```diff
@@ -16,58 +16,53 @@
 
 using System;
 using System.Diagnostics;
+using OpenTelemetry.AutoInstrumentation.Logging;
 using OpenTelemetry.AutoInstrumentation.Tagging;
 using OpenTelemetry.Trace;
 
 namespace OpenTelemetry.AutoInstrumentation.Util;
 
 internal static class ActivityHelper
 {
+    private static readonly ILogger Log = OtelLogging.GetLogger();
+
     /// <summary>
     /// Add the StackTrace and other exception metadata to the span
     /// </summary>
     /// <param name="activity">The activity to include exception info.</param>
     /// <param name="exception">The exception.</param>
     public static void SetException(this Activity activity, Exception exception)
     {
-        if (exception != null)
+        if (activity == null)
+        {
+            Log.Debug("Trying to set exception on null activity.");
+            return;
+        }
+
+        if (exception == null)
         {
-            activity.SetStatus(Status.Error.WithDescription(exception.Message));
-            activity.RecordException(exception);
+            return;
         }
+
+        activity.SetStatus(Status.Error.WithDescription(exception.Message));
+        activity.RecordException(exception);
     }
 
     public static Activity StartActivityWithTags(this ActivitySource activitySource, string operationName, ActivityKind kind, ITags tags)
     {
-        var activity = activitySource.StartActivity(operationName, kind);
+        var activity = activitySource?.StartActivity(operationName, kind);
```

Comment:
Do we want to log if `activitySource` is `null` in the same way as it is done in `SetException` extension method?